BUG: Fix build archs

carterbox opened the pull request:
On Windows, in the bld.bat, the extra archs for CUDA >=11.0 aren't actually being appended to that environment variable. This PR aims to fix this issue. Check the CI logs for the NVCC_FLAGS line of the CMAKE output to see if it is working correctly.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

carterbox committed:
BUG: Actually build for sm80,86

carterbox committed:
MNT: Re-rendered with conda-build 3.22.0, conda-smithy 3.21.2, and conda-forge-pinning 2022.10.26.19.07.37

carterbox committed:
Bump the build number

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

hmaarrfk commented:
hmm, i don't know too much about these things.

This website says that 11.1 should support 86 too.
https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/

carterbox commented on recipe/build.sh:
```diff
@@ -6,11 +6,12 @@ export PATH=$PREFIX/bin:$PATH
 
 export CUDA_ARCH_LIST="-gencode arch=compute_37,code=sm_37 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70"
```
Use this website for determining which archs to build for?

https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/

carterbox commented:
> This website says that 11.1 should support 86 too.

This build script agrees with [the website](https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/). It adds sm_86 for CUDA>=11.1.

carterbox committed:
BLD: Match cuda archs targeted in pytorch feedstock

This increases compatability on the low end and provides more
optimal code gen for intermediate archs like 61 and 75 WHICH includes
popular cards like to GTX 1000 series and RTX 2000 series cards.

https://github.com/conda-forge/pytorch-cpu-feedstock/blob/103634438261898bd3617336ad3594f831f31278/recipe/build_pytorch.sh

hmaarrfk commented:
> sm_86 for CUDA>=11.1.

I think i confused 11.0 with 11.1 when i first read your PR.
